src/devices/Ft4222/Ft4222Spi.cs

```diff
@@ -129,85 +134,53 @@ public Ft4222Spi(SpiConnectionSettings settings)
             return (FtClockRate.Clock80MHz, SpiClock.DivideBy1);
         }
 
-        /// <summary>
-        /// Reads data from the SPI device.
-        /// </summary>
-        /// <param name="buffer">
-        /// The buffer to read the data from the SPI device.
-        /// The length of the buffer determines how much data to read from the SPI device.
-        /// </param>
+        /// <inheritdoc/>
         public override void Read(Span<byte> buffer)
         {
-            byte[] readBuff = new byte[buffer.Length];
-            ushort readBytes = 0;
-            var ftStatus = FtFunction.FT4222_SPIMaster_SingleRead(_ftHandle, readBuff, (ushort)buffer.Length, ref readBytes, true);
+            ushort readBytes;
+            var ftStatus = FtFunction.FT4222_SPIMaster_SingleRead(_ftHandle, out MemoryMarshal.GetReference(buffer), (ushort)buffer.Length, out readBytes, true);
```

Comment:
buffer should be in/ref, I suspect out might work by accident too though

Reply:
Yes, all buffers are initialized before in fact, reason why it's working. out is allowing to pass something that is not initialized. Corrected for in everywhere.